common/sched.c

```diff
@@ -94,16 +97,33 @@ static task_t *create_task(void) {
     return task;
 }
 
-static void prepare_task(task_t *task, const char *name, task_func_t func, void *arg) {
-    if (!task)
+static void destroy_task(task_t **task) {
```

Comment:
I would make the input parameter just a pointer to task_t: `task_t *task` and add a comment that caller should set the parameter to NULL. It would make the code simpler here.